actionpack/lib/action_dispatch/routing/route_set.rb

```diff
@@ -513,6 +513,7 @@ def add_route(mapping, path_ast, name, anchor)
 
         route = @set.add_route(name, mapping)
         named_routes[name] = route if name
+        route.path.to_regexp
```

Comment:
As [I said on the original issue][1] there are a [bunch of instance variables][2] in `ActionDispatch::Journey::Path::Pattern` that could cause similar problems because they are lazily evaluated using `||=`. We should do as @rafaelfranca said and build them in the pattern's `initialize` method. Maybe even add a test to ensure that they're eagerly built but that might be too much for what is an internal implementation detail. However, definitely add a CHANGELOG entry.

[1]: https://github.com/rails/rails/issues/23699#issuecomment-184541398
[2]: https://github.com/rails/rails/blob/fe1f4b2/actionpack/lib/action_dispatch/journey/path/pattern.rb#L23-L27